### ISSUE TITLE
Nesting filters within feMerge

### ISSUE THREAD

**Issue by santoxi** [mkaemmerer/filter-effects#1]
Hi !
I'm trying hard to understand your monadic library :) 
I have a question: how would you nest nodes like feMergeNode within a feMerge element ?

result should be:

<filter id="feOffset" x="-40" y="-20" width="100" height="200">
    <feOffset in="SourceGraphic" dx="60" dy="60" />
    <feGaussianBlur in="SourceGraphic" stdDeviation="5" result="blur2" />
    <feMerge>
      <feMergeNode in="blur2" />
      <feMergeNode in="SourceGraphic" />
    </feMerge>
  </filter>

I have tried:

      const p=Filter.do(function* (){
        const x= yield filterEffects.sourceGraphic();
        const y= yield filterEffects.feGaussianBlur({in: x, stdDeviation: 5});

        const w = Filter.do(function* (){
          const x2= yield filterEffects.feMergeNode({in: y});
          const y2= yield filterEffects.feMergeNode({in: x});
          return Filter.done();
        })
        return Filter.done(w)
      })

but did not work :(

thanks !

**Comment by santoxi** [comment]
sorry but my xml was filtered out :( 
result should be:
`<filter id="feOffset" x="-40" y="-20" width="100" height="200">
    <feOffset in="SourceGraphic" dx="60" dy="60" />
    <feGaussianBlur in="SourceGraphic" stdDeviation="5" result="blur2" />
    <feMerge>
      <feMergeNode in="blur2" />
      <feMergeNode in="SourceGraphic" />
    </feMerge>
  </filter>`

**Comment by santoxi** [comment]
same happens with feComponentTransfer (which has nested feFuncR etc)

**Comment by mkaemmerer** [comment]
Nested nodes are definitely a weak point in this library right now. I would like to update this with a cleaner solution if I can find one I'm happy with.

For now, it **IS** possible to use nested nodes, but you need to make use of some undocumented information. You can create nested nodes by assigning to `children`. The value should be an array of objects describing the child nodes you want. For example:

```
const w = yield f.feMerge({
      children: [
        { nodeName: 'feMergeNode', in: x },
        { nodeName: 'feMergeNode', in: y },
      ]
    })
```

I've been reluctant to document this because it's a case where the library's internal representation is leaking out a bit. It's also often possible to achieve the same effects without nesting nodes. For example, using `feComposite` instead of `feMerge`.

Hope this helps!

**Comment by santoxi** [comment]
Hi again !
tried what you suggested, but did not work:

```
  const p=Filter.do(function* (){
    const x= yield filterEffects.sourceGraphic();
    const y= yield filterEffects.feGaussianBlur({in: x, stdDeviation: 5});

    const w = yield filterEffects.feMerge({
      children: [
        { nodeName: 'feMergeNode', in: x },
        { nodeName: 'feMergeNode', in: y },
      ]
    })

    return Filter.done(w)
  })

  const f = new Filter(p,{id: 'filtro', width: '200%', height: '200%' });

  $('#svg defs')[0].innerHTML = f.print();
```

which generates:

```
<filter id="filtro" width="200%" height="200%">
<feMerge children="[object Object],[object Object]" result="1"></feMerge>
</filter>
```

Do you have any other suggestion ?

Thanks :)

**Comment by santoxi** [comment]
Hi !

I just learned that, in order to have your trick work with nested children, I had to change src/print.js printNode function to this:

```
const printNode = node => {
  const {nodeName, children, ...attrs} = node;
  if(children){
    return `
      <${nodeName} ${printAttrs(attrs)}>
        ${children.map(printNode).join("\n")}
      </${nodeName}>
      `;
  }else{
    return `<${nodeName} ${printAttrs(attrs)}/>`;
  }
};
```

would you please make this change in src/print.js ?

thanks !

**Comment by mkaemmerer** [comment]
Thank you!
Yes, it looks like I made a mistake in `print` whereas in `build` children are handled correctly. I will make that change and publish a new version.

**Comment by mkaemmerer** [comment]
The new fixed print function is available! Please upgrade to version 0.8.1
